.github/workflows/zjit-macos.yml

```diff
@@ -35,6 +35,12 @@ jobs:
           - test_task: 'zjit-test'
             configure: '--enable-zjit=dev'
 
+          - test_task: 'ruby' # build test for combo build
```

Comment:
We should probably do this in `.github/workflows/compilers.yml`. It used to be just a huge matrix but now it allows us to minimize the number of jobs (because we have limited parallelism) for this kind of test.

Reply:
I moved the Ubuntu ones, but there's no macOS runs in `compilers.yml` yet, so I'm going to leave these.